src/selection.jl

```diff
@@ -144,6 +144,34 @@ function selectvalues(x::NDSparse, which; kwargs...)
     ndsparse(keys(x), rows(values(x), which); kwargs...)
 end
 
+struct OnAll{N, T<:Tuple}
+    functions::T
+    names::NTuple{N, Symbol}
+end
+
+OnAll(args...) = OnAll(args)
+OnAll(t::Tuple) = OnAll(t, map(Symbol,t))
+
+getselection(t, f) = f
+getselection(t, onall::OnAll) = Tuple(Symbol(s, :_, n) => s => f for s in colnames(t),
+    (f, n) in zip(onall.functions, onall.names))
+
+struct Not{T}
+    names::T
+end
+
+Not(args...) = Not(args)
+
+for f in [:(Base.select), :columns, :rows, :excludecols]
+    @eval $f(t::AbstractIndexedTable, not::Not) = $f(t, excludecols(t, not.names))
```

Comment:
Just defining `columns(x, ::Not)` should make `rows` work.

Reply:
I tried something like this a few days ago, I think the right way to do this would be to have something like this:

```julia
column(x, y::Union{Not, Keys}) = rows(x, lowerselection(x, y))
```
Where `lowerselection(x, y)` returns a tuple of column numbers...

This definition also answers this question: given
```julia
julia> t = table([1], [2], [3], names=[:foo,:bar,:baz])
Table with 1 rows, 3 columns:
foo  bar  baz
─────────────
1    2    3
```
What should `select(t, (:foo, Not(:bar))` do?

the options are:

1)
```julia
julia> select(t, (:foo, (:foo, :baz)))
Table with 1 rows, 2 columns:
1  2
─────────────────────
1  (foo = 1, baz = 3)
```
2)
```julia
julia> select(t, (:foo, :foo, :baz))
ERROR: syntax: duplicate field name: "foo" is not unique
```
I think it should be 1.